ods_ci/tasks/Resources/Provisioning/GPU/provision-gpu.sh

```diff
@@ -1,20 +1,72 @@
 #!/bin/bash
 set -e
 
+# GPU Node Provisioning Script for Self-Managed Clusters
+# 
+# This script creates GPU nodes in self-managed OpenShift clusters by creating
+# a new MachineSet with the specified configuration.
+#
+# Parameters:
+#   $1 INSTANCE_TYPE   - Instance type for the GPU nodes (default: g4dn.xlarge)
+#   $2 PROVIDER        - Cloud provider (AWS/GCP/AZURE/IBM) (default: AWS)
+#   $3 GPU_COUNT       - Number of GPUs per node (for GCP only) (default: 1)
+#   $4 GPU_NODE_COUNT  - Number of GPU nodes to create (default: 1)
+#
+# Examples:
+#   ./provision-gpu.sh                           # Create 1 g4dn.xlarge node on AWS
+#   ./provision-gpu.sh g4dn.2xlarge AWS 1 3     # Create 3 g4dn.2xlarge nodes on AWS
+#   ./provision-gpu.sh n1-standard-4 GCP 2 2    # Create 2 nodes with 2 GPUs each on GCP
+
 # Optional params
 INSTANCE_TYPE=${1:-"g4dn.xlarge"}
 PROVIDER=${2:-"AWS"}
 GPU_COUNT=${3:-"1"}
+GPU_NODE_COUNT=${4:-"1"}
 KUSTOMIZE_PATH="$PWD/tasks/Resources/Provisioning/Hive/GPU"
 MACHINESET_PATH="$KUSTOMIZE_PATH/base/source-machineset.yaml"
 PROVIDER_OVERLAY_DIR=$KUSTOMIZE_PATH/overlays/$PROVIDER
 MACHINE_WAIT_TIMEOUT=10m
 # Check if existing machineset GPU already exists
 EXISTING_GPU_MACHINESET="$(oc get machinesets.machine.openshift.io -n openshift-machine-api -o jsonpath="{.items[?(@.metadata.annotations['machine\.openshift\.io/GPU']>'0')].metadata.name}")"
 if [[ -n "$EXISTING_GPU_MACHINESET" ]] ; then
-  echo "Machine-set for GPU already exists"
-  oc get machinesets -A --show-labels
-  exit 0
+  echo "Machine-set for GPU already exists: $EXISTING_GPU_MACHINESET"
+  
+  # Get current replica count with proper error handling
+  CURRENT_REPLICAS=$(oc get machineset.machine.openshift.io $EXISTING_GPU_MACHINESET -n openshift-machine-api -o jsonpath='{.spec.replicas}' 2>/dev/null)
+  
+  # Check if we could get the replica count
+  if [[ -z "$CURRENT_REPLICAS" ]]; then
+    echo "Error: Could not get replica count for MachineSet $EXISTING_GPU_MACHINESET"
+    echo "MachineSet might not exist or use different API version. Listing all GPU MachineSets:"
+    oc get machinesets -n openshift-machine-api | grep -i gpu || echo "No GPU MachineSets found"
+    echo "Proceeding to create new MachineSet..."
+  else
+    echo "Current replicas: $CURRENT_REPLICAS, Desired replicas: $GPU_NODE_COUNT"
+  fi
+  
+  # Scale if needed (only if we successfully got the current replica count)
+  if [[ -n "$CURRENT_REPLICAS" && "$CURRENT_REPLICAS" != "$GPU_NODE_COUNT" ]]; then
```

Comment:
The feature of updating the replicas count in an existing Machine Set was not in scope of the ticket, but ok.

I think if we want to keep it, we should consider also:

- GPU flavor: here the code update the node count regardless the flavor of the GPU, hence if the script succeeds, the CI/CD user may think everything was OK, but actually they got a different flavor from the one they requested. What to do?
- What about the specific case of some NVIDIA GPUs on GCP where we need to specify the GPU count? This depends on how you're going to change the workflow on the CI side